### ISSUE TITLE
When defining onSelecting, the same slot can't be dragged over using a mouse or long pressed using touch events multiple times in a row.

### ISSUE THREAD

**Issue by brenthosie** [intljusticemission/react-big-calendar#858]
#### Do you want to request a _feature_ or report a _bug_?
report a _bug_

#### What's the current behavior?

When I use the `onSelecting`, `selectable`, and `onSelectSlot` props together with each other, I can't do a drag selection of just 1 slot or a long press of 1 slot on a touch device multiple times back-to-back and have the `onSelectSlot` callback fire.

![big-calendar-selection-bug](https://user-images.githubusercontent.com/1789219/40928243-10d811da-67de-11e8-9f21-30287328abe2.gif)

https://codesandbox.io/s/719rxl98m6

Browser: Chrome 66.0.3359.181

react-big-calendar @v0.19.1

#### What's the expected behavior?

When I use the `onSelecting`, `selectable`, and `onSelectSlot` props together with each other, I should be able to do a drag selection of just 1 slot or a long press of 1 slot on a touch device multiple times back-to-back and have the `onSelectSlot` callback fire.

**Comment by arecvlohe** [comment]
@brenthosie Are you saying you want to be to able to select and drag going from left to right instead of just top to bottom?

**Comment by brenthosie** [comment]
@arecvlohe no, the problem is that once you start a drag and only select 1 slot, then the callback of `onSelectSlot` stops firing reliably.

**Comment by arecvlohe** [comment]
Ah I see. @jquense will probably have more insight into this than I do.

**Comment by brenthosie** [comment]
@arecvlohe any word on this?
